trace/kafka/kafka.go

```diff
@@ -123,21 +127,33 @@ func (ap *AsyncProducer) propagateError() {
 	}
 }
 
-func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span) (*sarama.ProducerMessage, error) {
+func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span, enc encoding.EncodeFunc, ct string) (*sarama.ProducerMessage, error) {
 	c := kafkaHeadersCarrier{}
 	err := sp.Tracer().Inject(sp.Context(), opentracing.TextMap, &c)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to inject tracing headers")
 	}
-	var saramaKey sarama.Encoder
+
+	if len(ct) != 0 {
+		c.Set(encoding.ContentTypeHeader, ct)
+	}
+
+	var saramaKey, saramaBody sarama.Encoder
 	if msg.key != nil {
-		saramaKey = sarama.ByteEncoder(*msg.key)
+		saramaKey = sarama.StringEncoder(*msg.key)
 	}
+
+	b, err := enc(msg.body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to encode message body")
+	}
+	saramaBody = sarama.ByteEncoder(b)
```

Comment:
no need for this variable, we can add it directly to the below struct initialization